resources/builder.inc.sh

```diff
@@ -188,6 +188,9 @@ _builder_failure_trap() {
     fi
 
     builder_finish_action failure $action $target
+
+    # Ensure that the exit code propagates fully.
+    exit $trappedExitCode
   fi
```

Comment:
Can we pull this fix into a separate PR?

Reply:
See #7405.

Reply:
Thanks. Will review.